pants-plugins/src/python/internal_backend/sitegen/tasks/sitegen.py

```diff
@@ -45,6 +47,15 @@ def register_options(cls, register):
     super(SiteGen, cls).register_options(register)
     register('--config-path', type=list, help='Path to .json file describing site structure.')
 
+  # FIXME(cosmicexplorer): requiring these products ensures that the markdown
+  # and reference tasks run before this one, but we don't use those
+  # products.
+  @classmethod
+  def prepare(cls, options, round_manager):
```

Comment:
Nice!

Reply:
I didn't go all the way to do that here because I think sitegen.py might benefit from a complete revamp (which would result in cutting the line count way down and making sitegen far easier to understand and extend). Still need to iterate on that one but this is a good first step.